code-challenges.test.js

```diff
@@ -65,3 +159,18 @@ const accountTransactions3 = []
 // Expected output: []
 
 // b) Create the function that makes the test pass.
+// create a function "newArray" as well as a conditional for empty array
+// create conditional for numbers in array
+// create an empty variable "arr2"
+
+
+      const newArray = (object) => {
+        if (objet.length === null) {
+            return []
+        } else {
+          for (i=0, i <arr.length; i++;) {
+            let arr2= []
+            return arr2= [...arr2, (arr2 + arr[i])]
+          }
+        }
+      }
```

Comment:
I can tell you put a lot of time and thought into this one.  This is a tough one.  Though it's not working, I can see your thought process very well - which is great, by the way!  Just a note - an empty array is not the same as null.  :).